library/pod-security-policy/fsgroup/template.yaml

```diff
@@ -44,62 +44,76 @@ spec:
                     type: integer
   targets:
     - target: admission.k8s.gatekeeper.sh
-      rego: |
-        package k8spspfsgroup
+      code:
+      - engine: K8sNativeValidation
+        source:
+          variables:
+          - name: fsGroup
+            expression: '!has(variables.anyObject.spec.securityContext) ? "" : !has(variables.anyObject.spec.securityContext.fsGroup) ? "" : variables.anyObject.spec.securityContext.fsGroup'
+          - name: input_fsGroup_allowed
+            expression: |
+              !has(variables.params.rule) ? true : variables.params.rule == "RunAsAny" ? true : variables.params.rule == "MayRunAs" && variables.fsGroup == "" ? true : (variables.params.rule == "MayRunAs" || variables.params.rule == "MustRunAs") && has(variables.params.ranges) && size(variables.params.ranges) > 0 ? variables.params.ranges.all(range, range.min <= variables.fsGroup && range.max >= variables.fsGroup) : false
```

Comment:
It looks like the rego defaults to violating if `ranges` is an empty array. This is true for:

* MustRunAs
* MayRunAs when securityContext defines an fsGroup

It may be worth setting up a gator test case to validate these edge cases.